source/ui/BioXAS/BioXASSIS3820ScalerView.cpp

```diff
@@ -0,0 +1,133 @@
+
+	channelsBox_ = new QGroupBox();
+	channelsBox_->setTitle("Channels");
+
+	CLSSIS3820ScalerDarkCurrentWidget *darkCurrentWidget = new CLSSIS3820ScalerDarkCurrentWidget(scaler_);
+
+	darkCurrentBox_ = new QGroupBox();
+	darkCurrentBox_->setTitle("Dark current");
+
+	// Create and set layouts.
+
+	QHBoxLayout *controlsLayout = new QHBoxLayout();
+	controlsLayout->addStretch();
+	controlsLayout->addWidget(controlsView);
+	controlsLayout->addStretch();
+
+	controlsBox_->setLayout(controlsLayout);
+
+	QHBoxLayout *optionsLayout = new QHBoxLayout();
+	optionsLayout->addStretch();
+	optionsLayout->addWidget(biasEnabledShown);
+	optionsLayout->addWidget(biasShown);
+	optionsLayout->addWidget(darkCurrentShown);
+	optionsLayout->addStretch();
+
+	optionsBox_->setLayout(optionsLayout);
+
+	QVBoxLayout *channelsLayout = new QVBoxLayout();
+	channelsLayout->addWidget(channelsView);
+
+	channelsBox_->setLayout(channelsLayout);
+
+	QHBoxLayout *darkCurrentLayout = new QHBoxLayout();
+	darkCurrentLayout->addStretch();
+	darkCurrentLayout->addWidget(darkCurrentWidget);
+	darkCurrentLayout->addStretch();
+
+	darkCurrentBox_->setLayout(darkCurrentLayout);
+
+	QVBoxLayout *layout = new QVBoxLayout();
+	layout->addWidget(controlsBox_);
+	layout->addWidget(optionsBox_);
+	layout->addWidget(channelsBox_);
+	layout->addWidget(darkCurrentBox_);
+
+	setLayout(layout);
+
+	// Initial settings.
+
+	biasEnabledShown->setChecked(false);
+	biasShown->setChecked(false);
+	darkCurrentShown->setChecked(false);
+
+	optionsBox_->hide();
+	channelsBox_->hide();
+	darkCurrentBox_->hide();
+
+	// Make connections.
+
+	connect( biasEnabledShown, SIGNAL(clicked(bool)), channelsView, SLOT(setBiasEnabledEditorShown(bool)) );
+	connect( biasShown, SIGNAL(clicked(bool)), channelsView, SLOT(setBiasEditorShown(bool)) );
+	connect( darkCurrentShown, SIGNAL(clicked(bool)), channelsView, SLOT(setDarkCurrentShown(bool)) );
+	connect( darkCurrentShown, SIGNAL(clicked(bool)), this, SLOT(showDarkCurrentWidget(bool)) );
+
+	if (scaler_)
+		connect( scaler_, SIGNAL(connectedChanged(bool)), this, SLOT(showChannelViews(bool)) );
+
+	connect( channelsView, SIGNAL(biasEnabledEditorShownChanged(bool)), biasEnabledShown, SLOT(setChecked(bool)) );
+	connect( channelsView, SIGNAL(biasEditorShownChanged(bool)), biasShown, SLOT(setChecked(bool)) );
+
+	// Current settings.
+
+	biasEnabledShown->setChecked(channelsView->biasEnabledEditorShown());
+	biasShown->setChecked(channelsView->biasEditorShown());
+	darkCurrentShown->setChecked(channelsView->darkCurrentShown());
+
+	showOptions(optionsShown);
+
+	showChannelViews(true);
+}
+
+BioXASSIS3820ScalerView::~BioXASSIS3820ScalerView()
+{
+
+}
+
+void BioXASSIS3820ScalerView::showOptions(bool shown)
+{
+	optionsBox_->setShown(shown);
+}
+
+void BioXASSIS3820ScalerView::showChannelViews(bool shown)
+{
+	if (scaler_ && scaler_->isConnected() && shown)
```

Comment:
You could simplify this code with ```setVisible(scaler_ && scaler_->isConnected() && shown)``` and it would achieve the same thing.  Just mentioning it for interests sake.